portfolio/src/test/java/com/google/sps/ServletTest.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class ServletTest {
+
+    protected final LocalServiceTestHelper helper =
+            new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    protected HttpServletRequest request;
+    protected HttpServletResponse response;
+    protected PrintWriter writer;
+    protected StringWriter stringWriter;
+
+    @Before
+    public void setUp() throws IOException {
+        helper.setUp();
+
+        this.request = mock(HttpServletRequest.class);
+        this.response = mock(HttpServletResponse.class);
+        this.stringWriter = new StringWriter();
+        this.writer = new PrintWriter(this.stringWriter);
+        when(this.response.getWriter()).thenReturn(this.writer);
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    // Run this test twice to prove we're not leaking any state across tests.
+    private void doTest() {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        assertEquals(0, ds.prepare(new Query("yam")).countEntities(withLimit(10)));
+        ds.put(new Entity("yam"));
+        ds.put(new Entity("yam"));
+        assertEquals(2, ds.prepare(new Query("yam")).countEntities(withLimit(10)));
+    }
+
+    @Test
+    public void testInsert1() {
+        doTest();
+    }
+
+    @Test
+    public void testInsert2() {
+        doTest();
+    }
```

Comment:
I'm not sure what you are trying to test here. Not leaking state is a JUnit property that we wouldn't need to test ourselves. Maybe I'm misunderstanding what you're trying to achieve... Can you please give more context?